drag/drag.view.ts

```diff
@@ -1,41 +1,181 @@
+          draggedOverElem.parentElement === this.dragged_task!.parentElement
+        ) {
+          const parent = this.dragged_task!.parentElement;
+          const draggedOverIndex = Array.from(parent!.children).indexOf(
+            draggedOverElem as HTMLElement
+          );
+  
+          if (draggedOverIndex !== this.initial_task_index) {
+            parent!.insertBefore(this.dragged_task!, draggedOverElem);
+            this.initial_task_index = draggedOverIndex;
+          }
+        }
+      }
+  
+      drag_end(event: DragEvent | TouchEvent) {
+        if (event instanceof TouchEvent) {
+          this.handle_touch_end(event);
+          return;
+        }
+  
+        setTimeout(() => this.status("ready"));
+  
+        this.dragged_task = null;
+        this.initial_task_index = null;
+  
+        this.end(event as DragEvent);
+      }
+  
+      handle_touch_start(event: TouchEvent) {
+        event.preventDefault();
+        this.status("drag");
+  
+        this.dragged_task = event.target as HTMLElement;
+        this.initial_task_index = Array.from(
+          this.dragged_task!.parentElement!.children
+        ).indexOf(this.dragged_task);
+  
+        this.dragImageElem = this.dragged_task.cloneNode(true) as HTMLElement;
+  
+        this.dragImageElem.style.position = "absolute";
+        this.dragImageElem.style.pointerEvents = "none"; 
+        this.dragImageElem.style.zIndex = "9999";
+		this.dragImageElem.style.color = "var(--mol_theme_focus)";
+  
+        document.body.appendChild(this.dragImageElem);
+  
+        const touch = event.touches[0];
+        this.moveDragImage(touch.clientX, touch.clientY);
+  
+        this.start(event);
+      }
+  
+      handle_touch_move(event: TouchEvent) {
+        event.preventDefault();
+  
+        const touch = event.touches[0];
+        const draggedOverElem = document.elementFromPoint(
+          touch.clientX,
+          touch.clientY
+        );
+  
+        if (this.dragImageElem) {
+          this.moveDragImage(touch.clientX, touch.clientY);
+        }
+  
+        this.draggedOverElem = draggedOverElem as HTMLElement;
+  
+        if (
+          draggedOverElem &&
+          draggedOverElem !== this.dragged_task &&
+          draggedOverElem.parentElement === this.dragged_task!.parentElement
+        ) {
+          const parent = this.dragged_task!.parentElement;
+          const draggedOverIndex = Array.from(parent!.children).indexOf(
+            draggedOverElem as HTMLElement
+          );
+  
+          if (draggedOverIndex !== this.initial_task_index) {
+            parent!.insertBefore(this.dragged_task!, draggedOverElem);
+            this.initial_task_index = draggedOverIndex;
+          }
+        }
+      }
+  
+      handle_touch_end(event: TouchEvent) {
+        if (this.draggedOverElem) {
+          if (
+            this.draggedOverElem.id ==
+            "$hyoo_mol.Root(0).Demos().Widget('$mol_drag_demo').Trash()"
```

Comment:
This will only work for `$hyoo_mol.Root(0).Demos().Widget('$mol_drag_demo').Trash()`.

Anyway, this should be handled in `$mol_drop`, not `$mol_drag`.